src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -634,16 +635,16 @@ async Task<HttpResponseMessage> CreateHandlerAndSendAsync(HttpRequestMessage req
         {
             if (request.Version != HttpVersion.Version10 && request.Version != HttpVersion.Version11 && request.Version != HttpVersion.Version20 && request.Version != HttpVersion.Version30)
             {
-                return new NotSupportedException(SR.net_http_unsupported_version);
+                return ExceptionDispatchInfo.SetCurrentStackTrace(new NotSupportedException(SR.net_http_unsupported_version));
             }
 
             // Add headers to define content transfer, if not present
             if (request.HasHeaders && request.Headers.TransferEncodingChunked.GetValueOrDefault())
             {
                 if (request.Content == null)
                 {
-                    return new HttpRequestException(SR.net_http_client_execution_error,
-                        new InvalidOperationException(SR.net_http_chunked_not_allowed_with_empty_content));
+                    return ExceptionDispatchInfo.SetCurrentStackTrace(new HttpRequestException(SR.net_http_client_execution_error,
+                        ExceptionDispatchInfo.SetCurrentStackTrace(new InvalidOperationException(SR.net_http_chunked_not_allowed_with_empty_content))));
```

Comment:
I'm not sure if this is necessary as they both come from the same place. Note that we have similar thing in sockets, where we set the stack trace only to the inner one:
https://github.com/dotnet/runtime/blob/1d6452aedc5a2c74883916c8efc548a7cb51e902/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs#L1373-L1376

https://github.com/dotnet/runtime/blob/1d6452aedc5a2c74883916c8efc548a7cb51e902/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs#L897-L903

Reply:
It might be slightly better on just one since it avoids some duplicate text in the stack trace, but it shouldn't really be a problem.